src/sock.c

```diff
@@ -276,17 +276,21 @@ static void read_conn(struct mg_connection *c) {
     size_t len = c->recv.size - c->recv.len;
     long n = -1;
     if (c->is_tls) {
-      if (!ioalloc(c, &c->rtls)) return;
-      n = recv_raw(c, (char *) &c->rtls.buf[c->rtls.len],
-                   c->rtls.size - c->rtls.len);
-      if (n == MG_IO_ERR && c->rtls.len == 0) {
-        // Close only if we have fully drained both raw (rtls) and TLS buffers
-        c->is_closing = 1;
-      } else {
-        if (n > 0) c->rtls.len += (size_t) n;
-        if (c->is_tls_hs) mg_tls_handshake(c);
-        n = c->is_tls_hs ? (long) MG_IO_WAIT : mg_tls_recv(c, buf, len);
+      // Do not read to the raw TLS buffer if it already has enough.
+      // This is to prevent overflowing c->rtls if our reads are slow
+      if (c->rtls.len < 4096) {
```

Comment:
Where did this magic number come from ?
We have `MG_IO_SIZE` ruling that buffer resizing.
In fact, I addressed a couple of Unix issues/questions by contributing to our _best practices_ section:
> in scenarios where data accumulates faster than it can be consumed, as for example, in TLS large file transfers, you need to make MG_IO_SIZE larger to avoid reaching the MG_MAX_RECV_SIZE limit while buffering.

With this patch, we don't fill rtls, then data will accumulate in recv... but the user can throttle there catching EV_READ, right ?

So, should this be written as a function of `MG_IO_SIZE` or is it a fixed value related to TLS block size or something ?